ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -13,24 +80,255 @@ class RagasFaithfulnessEvaluator:
+                value=score_result_or_failure,
+            )
+
+    def evaluate(self, span_event: dict) -> Union[float, str]:
+        """
+        Performs a faithfulness evaluation on a span event, returning either
+            - faithfulness score (float) OR
+            - failure reason (str)
+        If the ragas faithfulness instance does not have `llm` set, we set `llm` using the `llm_factory()`
+        method from ragas which defaults to openai's gpt-4o-turbo.
+        """
+        if not self.ragas_dependencies_present:
+            return "fail_dependencies_not_present"
+
+        self.ragas_faithfulness_instance = _get_faithfulness_instance()
+
+        score, question, answer, context, statements, faithfulness_list = math.nan, None, None, None, None, None
+
+        with self.llmobs_service.workflow(
+            "ragas.faithfulness", ml_app=_get_ml_app_for_ragas_trace(span_event)
+        ) as ragas_faithfulness_workflow:
+            try:
+                faithfulness_inputs = self._extract_faithfulness_inputs(span_event)
+                if faithfulness_inputs is None:
+                    logger.debug(
+                        "Failed to extract question and context from span sampled for ragas_faithfulness evaluation"
+                    )
+                    return "fail_extract_faithfulness_inputs"
+
+                question = faithfulness_inputs["question"]
+                answer = faithfulness_inputs["answer"]
+                context = faithfulness_inputs["context"]
+
+                statements = self._create_statements(question, answer)
+                if statements is None:
+                    logger.debug("Failed to create statements from answer for `ragas_faithfulness` evaluator")
+                    return "statements_is_none"
+
+                faithfulness_list = self._create_verdicts(context, statements)
+                if faithfulness_list is None:
+                    logger.debug("Failed to create faithfulness list `ragas_faithfulness` evaluator")
+                    return "statements_create_faithfulness_list"
+
+                score = self._compute_score(faithfulness_list)
+                if math.isnan(score):
+                    logger.debug("Score computation returned NaN for `ragas_faithfulness` evaluator")
+                    return "statements_compute_score"
+
+                return score
+            finally:
+                self.llmobs_service.annotate(
+                    span=ragas_faithfulness_workflow,
+                    input_data=span_event,
+                    output_data=score,
+                    metadata={
+                        "statements": statements,
+                        "faithfulness_list": faithfulness_list.dicts() if faithfulness_list else None,
+                    },
+                )
+
+    def _create_statements(self, question: str, answer: str) -> Optional[List[str]]:
+        with self.llmobs_service.workflow("ragas.create_statements"):
```

Comment:
`dd-ragas.create_statements`?
- makes it clear it's from dd
- consistent with `dd-ragas` ml app prefix name
Worth noting though; the llm spans from ragas itself will still be auto-instrumented and named generic things. We could try to wrangle the names using annotation contexts, but that might be complicated since multiple nested spans are produced per `generate_text` call